### DISCUSSION TITLE
0 values converted to "", problem for PK links

### DISCUSSION THREAD

**Discussion by tstauch** [Budibase/budibase#4718] [Help]
I have an external table with a composite key (id,published).
Published can be 0 or 1

In Budibase, if published is 0, it displays as empty cell. (this is true for PKs and other fields)
The _id link (nice feature btw except that it) also uses empty string as the value.

So I am getting links to [id] if published is 0
But if published is 1, then _id links to [id,1] and it works great

So, I could probably do some fancy javascript to workaround this (great to have so many JavaScript options btw, tks!!) 
but it would be much better if 0 values displayed as 0 in the table, and then in the _id of course.

Or maybe I did something wrong in my external data source config? Can't find anything that maps 0 -> ""

**Comment by mjashanks** [comment]
Hey @tstauch when you say

> ...it displays as empty cell.

Where are you experiencing this? can you screenshot it? Definately sounds like a bug..

**Comment by tstauch** [comment]
Here you will see the actual data (lower window) which includes the 0 values. But Budibase has empty cells.

The problem with this is that any links or logic using _id fails. It should be passing [id,0] if the "published" column is 0

![buildabase-0-problem](https://user-images.githubusercontent.com/4955990/156258084-5bf5c2ea-34f3-4c27-b14a-de90d0202e9d.png)

**Reply by mjashanks** [reply]
Thanks for this - I'll investigate this. Looks like Budibase thinks `published` is `undefined` when it is `0`. 

You could potentially use javascript to check if `_id.length === 2` etc... but I know that this could get annoying to deal with.

Will get this logged as an issue once I've recreated it.
